projects/plugins/jetpack/extensions/blocks/tiled-gallery/utils/index.js

```diff
@@ -112,9 +117,14 @@ export function photonizedImgProps( img, galleryAtts = {} ) {
 }
 function isVIP() {
 	/*global jetpack_plan*/
-	if ( typeof jetpack_plan !== 'undefined' && jetpack_plan.data === 'vip' ) {
-		return true;
+	// Use `jetpackPlanFromState` if available, otherwise fall back to `jetpack_plan` defined within the render function in tiled-gallery.php.
+	let jetpackPlan;
+	if ( typeof jetpackPlanFromState !== 'undefined' ) {
+		jetpackPlan = jetpackPlanFromState;
+	} else if ( typeof jetpack_plan !== 'undefined' ) {
+		jetpackPlan = jetpack_plan;
 	}
+	return jetpackPlan && jetpackPlan.data === 'vip';
```

Comment:
Do you think we should make sure `data` property exists?
```suggestion
	return jetpackPlan && jetpackPlan?.data === 'vip';
```

Reply:
Good idea.